src/NzbDrone.Core/Music/Services/TrackService.cs

```diff
@@ -115,6 +118,10 @@ public void UpdateMany(List<Track> tracks)
         public void DeleteMany(List<Track> tracks)
         {
             _trackRepository.DeleteMany(tracks);
+            foreach (var track in tracks)
+            {
+                _eventAggregator.PublishEvent(new TrackDeletedEvent(track));
```

Comment:
Rework so we are not throwing track delete events everywhere... (could be 1000s per artist delete). We only need them on a track remove from mb or a release remove, merges should have history migrated, album and artist deletions will handle history events for there child tracks.